localstripe/resources.py

```diff
@@ -772,6 +774,9 @@ def __init__(self, name=None, description=None, email=None,
         self.shipping = None
         self.default_source = None
 
+        if payment_method is not None:
+            PaymentMethod._api_attach(payment_method, customer=self.id)
```

Comment:
Just a final request: can you check the `payment_method` we try to attach really exists before?
Something like we do here for the `invoice` - just before the `super().__init__()`:
https://github.com/adrienverge/localstripe/blob/3a13cf8301ef43a25377e9c16b3d959577207999/localstripe/resources.py#L1564-L1565

Reply:
I have found that the endpoint already seems to return a 404 when provided with an invalid `payment_method`:
```bash
# creating a customer with a non-existant payment_method returns a 404                                
code=$(curl -sg -o /dev/null -w '%{http_code}' -u $SK: $HOST/v1/customers \                           
            -d description='This customer will have a payment_method when created' \                   
            -d payment_method='pm_doesnotexist')                                                       

[ "$code" -eq 404 ]  
```

Would you like me to add this test?

Does it still seem worth adding the check?

Reply:
> I have found that the endpoint already seems to return a 404 when provided with an invalid payment_method

Yes correst, `_api_attach()` returns 404 but the goal of adding this check is to avoid creating the customer if the payment method provided doesn't exist.

Reply:
Ah! That makes sense, thanks. OK, I'll get on that.

Reply:
Rebased and pushed :)

Reply:
Looks almost perfect. It would be better to differentiate the payment method check which returns 404 and the input values checks which returns 400 error.
Can you update the code so it looks like this:
```py
try:
    ...
    if payment_method is not None:
        assert type(payment_method) is str
except AssertionError:
    raise UserError(400, 'Bad request')

if payment_method is not None:
    PaymentMethod._api_retrieve(payment_method)  # to return 404 if not existant

# All exceptions must be raised before this point.
super().__init__()

self.name = name
...

if payment_method is not None:
    PaymentMethod._api_attach(payment_method, customer=self.id)
```
The test you added is nice!

Reply:
> It would be better to differentiate the payment method check which returns 404 and the input values checks which returns 400 error.

Good call. Done. 

> The test you added is nice!

Thanks!